CMakeLists.txt

```diff
@@ -92,13 +95,21 @@ function(set_default_compile_options target)
         endif()
         if(CMAKE_BUILD_TYPE STREQUAL "Debug")
             set(MAIN_CFLAGS "${MAIN_CFLAGS} -D_DEBUG")
+# code coverage generation using gcov and lcov - linux only
+            if (NOT APPLE AND ENABLE_COVERAGE)
+#                enable_testing()
+                # set compiler flags
+#                set(MAIN_CFLAGS "${MAIN_CFLAGS}-g -O0 -fprofile-arcs -ftest-coverage")
+                set(MAIN_CFLAGS " -D__UNIX__ -D__64BIT__ -g -O0 --coverage")
+           endif()            
```

Comment:
`D__UNIX__` is already set in line 91/94, `D__64BIT__` in line 78/81, do they need to be repeated?

Also, are the commented out code needed?

Reply:
For code coverage, the previously defined ${MAIN_CFLAGS} are overwritten with optimization removed, so the compiler flags -D__UNIX__ -D__64BIT__  are repeated for the coverage build.